docusaurus/docs/dev-docs/backend-customization.md

```diff
@@ -4,31 +4,87 @@ description: All elements of Strapi's back end, like routes, policies, middlewar
+The [User Guide](/user-docs/intro) explains how to use the admin panel and the [admin panel customization section](/dev-docs/admin-panel-customization) details the various customization options available for the admin panel.
+:::
 
-- the [responses](/dev-docs/backend-customization/requests-responses#responses) sent to the application that sent the request,
+The Strapi back end runs an HTTP server based on [Koa](https://koajs.com/), a back-end JavaScript framework.
+
+Like any HTTP server, the Strapi back end receives requests and send responses. You can send requests to the Strapi back end to create, retrieve, update, or delete data through the [REST](/dev-docs/api/rest) or [GraphQL](/dev-docs/api/graphql) APIs.
+
+A request can travel through the Strapi backend as follows:
+
+1. The Strapi server receives a [request](/dev-docs/backend-customization/requests-responses#requests).
+2. The request hits [global middlewares](/dev-docs/backend-customization/middlewares) that are run in a sequential order.
+3. The request hits a [route](/dev-docs/backend-customization/routes).<br/>By default, Strapi generates route files for all the content-types that you create (see [REST API documentation](/dev-docs/api/rest)), and more routes can be added and configured.
+4. [Route policies](/dev-docs/backend-customization/policies) act as a read-only validation step that can block access to a route. [Route middlewares](/dev-docs/backend-customization/routes#middlewares) can control the request flow and mutate the request itself before moving forward.
+5. [Controllers](/dev-docs/backend-customization/controllers) execute code once a route has been reached. [Services](/dev-docs/backend-customization/services) are optional, additional code that can be used to build custom logic reusable by controllers.
+6. The code executed by the controllers and services interacts with the [models](/dev-docs/backend-customization/models) that are a representation of the content data structure stored in the database.<br />Interacting with the data represented by the models is handled by the [Entity Service](/dev-docs/api/entity-service) and [Query Engine](/dev-docs/api/query-engine).
+7. The server returns a [response](/dev-docs/backend-customization/requests-responses#response). The response can travel back through route middlewares and global middlewares before being sent.
+
+Both global and route middlewares include an asynchronous callback function, `await next()`. Depending on what is returned by the middleware, the request will either go through a shorter or longer path through the backend:
+
+* If a middleware returns nothing, the request will continue travelling through the various core elements of the backend (i.e., controllers, services, and the other layers that interact with the database).
+* If a middleware returns something, a response will be immediately sent, skipping the rest of the core elements.
```

Comment:
```suggestion
* If a middleware returns before calling `await next()`, a response will be immediately sent, skipping the rest of the core elements. Then it will go back down the same chain it came up.
```